libopae/plugins/xfpga/sysfs.c

```diff
@@ -693,6 +708,110 @@ fpga_result sysfs_get_interface_id(fpga_token token, fpga_guid guid)
 }
 
 
+
+fpga_result sysfs_get_fme_pwr_path(fpga_token token, char *sysfs_pwr)
+{
+	fpga_result res = FPGA_OK;
+	struct _fpga_token *_token = (struct _fpga_token *)token;
+	ASSERT_NOT_NULL(_token);
+
+	if (sysfs_pwr == NULL) {
+		FPGA_ERR("Invalid input parameters");
+		return FPGA_INVALID_PARAM;
+	}
+	res = cat_token_sysfs_path(sysfs_pwr, token, SYSFS_FORMAT(sysfs_fme_pwr_glob));
+
+	return res;
+}
+
+fpga_result sysfs_get_fme_temp_path(fpga_token token, char *sysfs_temp)
+{
+	fpga_result res = FPGA_OK;
+	struct _fpga_token *_token = (struct _fpga_token *)token;
+	ASSERT_NOT_NULL(_token);
+
+	if (sysfs_temp == NULL) {
+		FPGA_ERR("Invalid input parameters");
+		return FPGA_INVALID_PARAM;
+	}
+
+	res = cat_token_sysfs_path(sysfs_temp, token, SYSFS_FORMAT(sysfs_fme_temp_glob));
+
+	return res;
+}
+
+fpga_result sysfs_get_fme_perf_path(fpga_token token, char *sysfs_perf)
+{
+	fpga_result res = FPGA_OK;
+	struct _fpga_token *_token = (struct _fpga_token *)token;
+	ASSERT_NOT_NULL(_token);
+
+	if (sysfs_perf == NULL) {
+		FPGA_ERR("Invalid input parameters");
+		return FPGA_INVALID_PARAM;
+	}
+
+	res = cat_token_sysfs_path(sysfs_perf, token, SYSFS_FORMAT(sysfs_fme_perf_glob));
```

Comment:
Shouldn't we resolve the glob pattern here?

Reply:
Metrics API glab enums folder with search pattern power*_*
Upstream driver TEMP sysfs is different from internal driver 


[lab@PSERA2-NCF22 mybuild]$ ls -l /sys/class/fpga_region/region0/dfl-fme.0/hwmon/hwmon3/
total 0
lrwxrwxrwx 1 root root    0 Nov 12 15:49 device -> ../../../dfl-fme.0
-r--r--r-- 1 root root 4096 Nov 12 15:49 name
drwxr-xr-x 2 root root    0 Nov 12 15:49 power
lrwxrwxrwx 1 root root    0 Nov 12 15:49 subsystem -> ../../../../../../../../class/hwmon
-r--r--r-- 1 root root 4096 Nov 12 15:49 temp1_crit
-r--r--r-- 1 root root 4096 Nov 12 15:49 temp1_crit_alarm
-r--r--r-- 1 root root 4096 Nov 12 15:49 temp1_emergency
-r--r--r-- 1 root root 4096 Nov 12 15:49 temp1_input
-r--r--r-- 1 root root 4096 Nov 12 15:49 temp1_max
-r--r--r-- 1 root root 4096 Nov 12 15:49 temp1_max_alarm
-r--r--r-- 1 root root 4096 Nov 12 15:49 temp1_max_policy
-rw-r--r-- 1 root root 4096 Nov 12 15:46 uevent